### PR TITLE
BACK-1535: Integrate sDAI

### PR DESCRIPTION
**Pull request by Wertusser**

### CODE REVIEW

File: src/abi/maker-psm/pot.json
Comment (JackieJoo) [inline src/abi/maker-psm/pot.json:1]: shouldn't it live inside sdai folder?

Comment (Wertusser) [reply src/abi/maker-psm/pot.json:1]: Pot is a part of Maker Protocol, thought it better to leave it there

File: src/dex/sdai/config.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { SDaiParams } from './types';
+import { DexConfigMap } from '../../types';
+import { Network } from '../../constants';
+import { SwapSide } from '@paraswap/core';
+
+export const SDaiConfig: DexConfigMap<SDaiParams> = {
+  sdai: {
```
Comment (JackieJoo) [inline src/dex/sdai/config.ts:7]: Rename to Spark?
https://docs.spark.fi/

File: src/dex/sdai/sdai-pool.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+export class SDaiEventPool extends StatefulEventSubscriber<SDaiPoolState> {
+  constructor(
+    parentName: string,
+    protected dexHelper: IDexHelper,
+    private potAddress: Address,
+    private potInterface: Interface,
+    logger: Logger,
+  ) {
+    super(parentName, 'sdai', dexHelper, logger);
```
Comment (JackieJoo) [inline src/dex/sdai/sdai-pool.ts:62]: Rename to Spark?
https://docs.spark.fi/

**Review by JackieJoo — APPROVED** [review]

**Review by nzrii-kysel — APPROVED** [review]
lgtm 👍

File: src/dex/spark/spark-sdai-pool.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+export class SparkSDaiEventPool extends StatefulEventSubscriber<SparkSDaiPoolState> {
+  constructor(
+    parentName: string,
+    protected dexHelper: IDexHelper,
+    private potAddress: Address,
+    private potInterface: Interface,
+    logger: Logger,
+  ) {
+    super(parentName, 'Spark', dexHelper, logger);
```
Comment (JackieJoo) [inline src/dex/spark/spark-sdai-pool.ts:62]: Please change the name to have token addresses inside

**Review by KanievskyiDanylo — APPROVED** [review]

**Pull request merged**